### ISSUE TITLE
[apex] ApexCRUDViolation doesn't report SOQL for loops

### ISSUE THREAD

**Issue by jonathanwiesel** [pmd/pmd#3329]
**Affects PMD Version:** 
5.5.3 - 6.35.0

**Rule:**
[ApexCRUDViolation](https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**

When using [SOQL For loops](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_loops_for_SOQL.htm) the CRUD check seems to be not enforced

**Code Sample demonstrating the issue:**

```
// won't report it
for (Account acc : [SELECT Id FROM Account]) {
  //whatever
}

// will report it
Account[] accs = [SELECT Id FROM Account];
```

**Expected outcome:**

PMD should report a violation at line 1, but doesn't. This is a false-negative.

**Running PMD through:** *Apex PMD VSCode Plugin*
